src/fault/Trigger.cpp

```diff
@@ -62,41 +62,61 @@ void Trigger::ensure(Type type) const
     if (type_ != type)
     {
         etiss::log(etiss::FATALERROR, std::string("etiss::fault::Trigger::ensure: Type mismatch type=") +
-                                          std::to_string(type) + " type_=" + std::to_string(type_));
+                                          type_t::toString(type) + " type_=" + std::string(type_));
         throw "called function of different trigger type";
     }
 }
 
-Trigger::Trigger() : type_(NOP)
+Trigger::Trigger()
+  : type_(Type::NOP)
 {
     etiss::log(etiss::VERBOSE, std::string("etiss::fault::Trigger::Trigger() : type_ (NOP)"));
 }
+
+Trigger::Trigger(const InjectorAddress &target_injector)
+  : type_(Type::ASAP)
+  , inj_(target_injector)
+{
+    etiss::log(etiss::VERBOSE, std::string("etiss::fault::Trigger::Trigger(const InjectorAddress &target_injector) : type_ (ASAP)"));
+}
+
 Trigger::Trigger(const Trigger &sub, uint64_t count)
-    : type_(META_COUNTER), sub_(new Trigger(sub)), param1_(count), param2_(0)
+  : type_(Type::META_COUNTER)
+  , sub_(new Trigger(sub))
+  , param1_(count)
+  , param2_(0)
```

Comment:
there are many uninitialized fields in these constructors. maybe just default initialize them in the class definition

Reply:
Fixed with https://github.com/tum-ei-eda/etiss/commit/909d74ea2a00d5ff3d42641f5e8b01a86f44e403